operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -135,8 +134,67 @@ func deprecatedSettingsMessage(iop *v1alpha1.IstioOperatorSpec) string {
 }
 
 // validateFeatures check whether the config sematically make sense. For example, feature X and feature Y can't be enabled together.
-func validateFeatures(_ *valuesv1alpha1.Values, _ *v1alpha1.IstioOperatorSpec) util.Errors {
-	return nil
+func validateFeatures(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) util.Errors {
+	return CheckServicePorts(values, spec)
+}
+
+func CheckServicePorts(values *valuesv1alpha1.Values, spec *v1alpha1.IstioOperatorSpec) util.Errors {
+	var errs util.Errors
+	if !values.GetGateways().GetIstioIngressgateway().GetRunAsRoot().GetValue() {
+		errs = util.AppendErrs(errs, validateGateways(spec.GetComponents().GetIngressGateways(), "istio-ingressgateway"))
+	}
```

Comment:
should this validate user gateways with different names too? also, i wonder if the name const could be derived rather than hardcoded here.

Reply:
It does, `spec.GetComponents().GetIngressGateways()` is a list of gateways. The name const is referring to the `values` name - NOT the gateway name. runAsRoot is part of values